mara_pipelines/commands/files.py

```diff
@@ -57,21 +57,37 @@ def db_alias(self):
         return self._db_alias or config.default_db_alias()
 
     def shell_command(self):
-        copy_from_stdin_command = mara_db.shell.copy_from_stdin_command(
-            self.db_alias(), csv_format=self.csv_format, target_table=self.target_table,
-            skip_header=self.skip_header,
-            delimiter_char=self.delimiter_char, quote_char=self.quote_char,
-            null_value_string=self.null_value_string, timezone=self.timezone)
-        if not isinstance(mara_db.dbs.db(self.db_alias()), mara_db.dbs.BigQueryDB):
-            return \
-                f'{uncompressor(self.compression)} "{pathlib.Path(config.data_dir()) / self.file_name}" \\\n' \
-                + (f'  | {shlex.quote(sys.executable)} "{self.mapper_file_path()}" \\\n'
-                   if self.mapper_script_file_name else '') \
-                + ('  | sort -u \\\n' if self.make_unique else '') \
-                + '  | ' + copy_from_stdin_command
+        file_path = pathlib.Path(config.data_dir()) / self.file_name
+        db = mara_db.dbs.db(self.db_alias())
+        if isinstance(db, mara_db.dbs.SQLServerDB):
+            if self.compression != Compression.NONE:
+                raise ValueError('A compression for ReadFile is not supported when the db_alias referres to a SQL Server (SQLServerDB)')
+            if self.csv_format == False:
+                raise ValueError('The parameter csv_format must be true or none when the db_alias referres to a SQL Server (SQLServerDB)')
+            return (f'bcp {self.target_table} in {file_path}'
```

Comment:
I remember bcp had a bug with big files where it missed some rows at the end of the file. Is that fixed nowadays?

https://feedback.azure.com/forums/908035-sql-server/suggestions/32895967-bcp-uploads-not-all-rows-from-a-csv-file

Reply:
I am not aware of this issue. I'd never run in an issue as you mentioned. Maybe better to split the source files in such cases.

When we look at bq extract for Google BigQuery, you can see [here](https://cloud.google.com/bigquery/docs/exporting-data#exporting_data_into_one_or_more_files) that even Google does not support files bigger than 1 GB. They added functions to split them during the export. In the bug report mentioned above, the file had a size of 1.2 GB.

Maybe we should add a recommendation to have max. 1 GB of (uncompressed) data?